src/System.Console/tests/WindowAndCursorProps.cs

```diff
@@ -33,8 +33,16 @@ public static void BufferSize_GettingSameAsWindowSize()
     [Fact]
     public static void WindowWidth_WindowHeight_InvalidSize()
     {
-        Assert.Throws<ArgumentOutOfRangeException>("width", () => Console.WindowWidth = 0);
-        Assert.Throws<ArgumentOutOfRangeException>("height", () => Console.WindowHeight = 0);
+        if (Console.IsOutputRedirected)
```

Comment:
The setter of both properties call the other property getter when calling `SetWindowSize` and that calls into `GetBufferInfo` which does the output redirection check.